src/apply_rules.rs

```diff
@@ -149,6 +152,28 @@ fn check_networks_consolidation(
     Ok(res)
 }
 
+fn update_ticket_use_perimeters(
+    collections: &mut Collections,
+    networks_consolidation: Vec<NetworkConsolidation>,
+) {
+    for network in networks_consolidation {
+        let network_id = network.network.id.clone();
+        for grouped_from in network.grouped_from {
+            let ticket_use_perimeter_by_network: Vec<&mut TicketUsePerimeter> = collections
+                .ticket_use_perimeters
+                .values_mut()
+                .filter(|ticket| {
+                    ticket.object_type == ModelObjectType::Network
+                        && ticket.object_id == grouped_from
+                })
+                .collect();
```

Comment:
hum I don't think you need the `collect` here.

you can either do:

```rust 
for mut ticket in collections
                .ticket_use_perimeters
                .values_mut()
                .filter(|ticket| {
                    ticket.object_type == ModelObjectType::Network
                        && ticket.object_id == grouped_from
                }) {
    ticket.object_id = network_id.to_string();
}
```

or maybe (not tested though)
```rust 
collections
                .ticket_use_perimeters
                .values_mut()
                .filter(|ticket| {
                    ticket.object_type == ModelObjectType::Network
                        && ticket.object_id == grouped_from
                }).for_each(|mut ticket| {
    ticket.object_id = network_id.to_string();
})
```